src/System.Security.Cryptography.Pkcs/tests/EnvelopedCms/KeyAgreeRecipientInfoTests.cs

```diff
@@ -297,6 +284,45 @@ public static void TestKeyAgreeOtherKeyAttribute_FixedValue_Ski()
             Assert.Equal<byte>(expectedAsnData, asnData.RawData);
         }
 
+        [Fact]
+        public static void TestMultipleKeyAgree_ShortNotation()
+        {
+            // KeyAgreement recipients are defined in RFC 2630 as
+            //
+            // KeyAgreeRecipientInfo::= SEQUENCE {
+            //      version CMSVersion,  --always set to 3
+            //      originator[0] EXPLICIT OriginatorIdentifierOrKey,
+            //      ukm[1] EXPLICIT UserKeyingMaterial OPTIONAL,
+            //      keyEncryptionAlgorithm KeyEncryptionAlgorithmIdentifier,
+            //      recipientEncryptedKeys RecipientEncryptedKeys }
+            //
+            // RecipientEncryptedKeys::= SEQUENCE OF RecipientEncryptedKey
+            //
+            // RecipientEncryptedKey::= SEQUENCE {
+            //      rid KeyAgreeRecipientIdentifier,
+            //      encryptedKey EncryptedKey }
+            //
+            // When RecipientEncryptedKeys has more than one sequence in it, then different KeyAgreement recipients are created, where each
+            // recipient hold the information from one RecipientEncryptedKey  
+
+            byte[] encodedMessage =
+                ("3082019206092A864886F70D010703A08201833082017F0201023182014BA1820147020103A08196A18193300906072A"
+                + "8648CE3E02010381850002818100AC89002E19D3A7DC35DAFBF083413483EF14691FC00A465B957496CA860BA4918182"
+                + "1CAFB50EB25330952BB11A71A44B44691CF9779999F1115497CD1CE238B452CA95622AF968E39F06E165D2EBE1991493"
+                + "70334D925AA47273751AC63A0EF80CDCF6331ED3324CD689BFFC90E61E9CC921C88EF5FB92B863053C4C1FABFE15301E"
+                + "060B2A864886F70D0109100305300F060B2A864886F70D010910030605003081883042A016041410DA1370316788112E"
+                + "B8594C864C2420AE7FBA420428DFBDC19AD44063478A0C125641BE274113441AD5891C78F925097F06A3DF57F3F1E6D1"
+                + "160F8D3C223042A016041410DA1370316788112EB8594C864C2420AE7FBA420428DFBDC19AD44063478A0C125641BE27"
+                + "4113441AD5891C78F925097F06A3DF57F3F1E6D1160F8D3C22302B06092A864886F70D010701301406082A864886F70D"
+                + "030704088AADC286F258F6D78008FC304F518A653F83").HexToByteArray();
+
+            EnvelopedCms ecms = new EnvelopedCms();
+            ecms.Decode(encodedMessage);
+
+            RecipientInfoCollection recipients = ecms.RecipientInfos;
+            Assert.Equal(2, recipients.Count);
+        }
```

Comment:
Check the type? And other metadata?  Just to be sure that both implementations are reading the metadata the same in the plural case.